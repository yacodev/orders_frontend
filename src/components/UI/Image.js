import styled from "@emotion/styled";
import PropTypes from "prop-types";

const IMAGE_SIZES = {
  sm: "62px",
  md: "130px",
  lg: "241px",
};

export const Image = styled.img`
  background: #ffffff;
  box-shadow: 0px 20px 20px rgba(0, 0, 0, 0.2);

  border-radius: 50%;
  width: ${(props) => IMAGE_SIZES[props.size] || IMAGE_SIZES.md};
  height: ${(props) => IMAGE_SIZES[props.size] || IMAGE_SIZES.md};
  object-fit: cover;
`;

Image.propTypes = {
  size: PropTypes.oneOf(["sm", "md", "lg"]),
};
