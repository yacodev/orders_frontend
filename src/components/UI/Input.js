import styled from "@emotion/styled";

const InputText = styled.input`
  border: none;
  border-bottom: 1px solid #333333;
  width: 100%;
  background: #f6f6f9;
`;

const Label = styled.label`
  font-size: 14px;
  line-height: 17px;
  color: #b8b8bb;
`;

const ContainerInput = styled.div`
  width: ${(props) => props.width};
  margin-top: ${(props) => props.marginTop};
`;

export const Input = (props) => {
  return (
    <ContainerInput width={props.width} marginTop={props.marginTop}>
      <Label htmlFor={props.name}>{props.label}</Label>
      <InputText
        type={props.type}
        name={props.name}
        value={props.value}
        onChange={props.onChange}
        disabled={props.disabled}
        required={props.required}
      />
    </ContainerInput>
  );
};
