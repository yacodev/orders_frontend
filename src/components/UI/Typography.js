import styled from "@emotion/styled";


export const Text = styled.p`
  color: ${props => props.color || "#000000"};
`
export const TextWhite = styled.p`
  color: #FFF;
`

export const RegularXL = styled(Text)`
  font-size: 28px;
  line-height: 35px;  
`;

export const RegularL = styled(Text)`
  font-size: 22px;
  line-height: 28px;
`;

export const RegularM = styled(Text)`
  font-size: 18px;
  line-height: 23px;
`;

export const RegularMWhite = styled(TextWhite)`
  font-size: 18px;
  line-height: 23px;
`;

export const RegularS = styled(Text)`
  font-size: 16px;
  line-height: 20px;
`;

export const RegularXS = styled(Text)`
  font-size: 15px;
  line-height: 18px;
`;

export const RegularCaption = styled(Text)`
  font-family: Lato;
  font-size: 14px;
  line-height: 17px;
`;

export const SemiBoldXL = styled(Text)`
  font-weight: 600;
  font-size: 28px;
  line-height: 35px;
`;

export const SemiBoldL = styled(Text)`
  font-weight: 600;
  font-size: 22px;
  line-height: 28px;
`;

export const SemiBoldM = styled(Text)`
  font-weight: 600;
  font-size: 18px;
  line-height: 23px;
`;

export const SemiBoldS = styled(Text)`
  font-weight: 600;
  font-size: 16px;
  line-height: 20px;
`;

export const SemiBoldCaption = styled(Text)`
  font-family: Lato;
  font-weight: 600;
  font-size: 14px;
  line-height: 17px;
`;


