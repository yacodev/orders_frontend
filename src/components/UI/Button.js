import styled from "@emotion/styled";
import { RegularM } from "./Typography";

const ButtonBase = styled.button`
  border-radius: 30px;
  border: none;
  width: 310px;
  height: 70px;
  margin-top: ${(props) => props.marginTop};
  cursor: ${(props) => (props.disabled ? "not-allowed" : "pointer")};
  background: ${(props) => (props.disabled ? "#B8B8BB" : "#FA4A0C")};
  .text--button {
    color: white;
  }
`;

export const Button = ({onClick, disabled, marginTop, text}) => {
  return (
    <ButtonBase
      onClick={onClick}
      disabled={disabled}
      marginTop={marginTop}
    >
      <RegularM className="text--button">{text}</RegularM>
    </ButtonBase>
  );
};
