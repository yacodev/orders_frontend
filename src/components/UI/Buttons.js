import styled from "@emotion/styled";
import { RegularM } from "./Typography";

const ButtonBase = styled.button`
  border-radius: 30px;
  border: none;
  width: 310px;
  height: 70px;
  margin-top: ${(props) => props.marginTop};
  cursor: ${(props) => (props.disabled ? "not-allowed" : "pointer")};
  background: ${(props) => (props.disabled ? "#B8B8BB" : "#FA4A0C")};
  .text--button {
    color: white;
  }
`;

export const Button = (props) => {
  return (
    <ButtonBase
      onClick={props.onClick}
      disabled={props.disabled}
      marginTop={props.marginTop}
    >
      <RegularM className="text--button">{props.text}</RegularM>
    </ButtonBase>
  );
};
