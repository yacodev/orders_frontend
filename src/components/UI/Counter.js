import styled from "@emotion/styled";
import { RegularMWhite } from "./Typography";
import { ADD_ITEM, SUBSTRACT_ITEM } from "../../constants";

export const CounterContainer = styled.div`
  background-color: #fa4a0c;
  border-radius: 30px;
  width: 52px;
  height: 20px;
  display: flex;
  flex-direction: row;
  color: #fff;
  justify-content: space-evenly;
  align-items: center;
  overflow: hidden;
`;

export const Symbol = styled.button`
  font-weight: 400;
  line-height: 23px;
  font-size: 20px;
  color: white;
  border: none;
  background: none;
  cursor: pointer;
`;

export const Counter = ({ count, updateCount }) => {
  return (
      <CounterContainer>
        <Symbol onClick={() => updateCount(SUBSTRACT_ITEM)}> - </Symbol>
        <RegularMWhite> {count} </RegularMWhite>
        <Symbol onClick={() => updateCount(ADD_ITEM)}> + </Symbol>
      </CounterContainer>
  );
};
