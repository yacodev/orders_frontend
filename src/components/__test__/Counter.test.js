import renderer from "react-test-renderer";
import { Counter } from "../UI/Counter";

describe("Counter", () => {
  it("renders correctly", () => {
    const count = 1;
    const setCount = () => {};
    const tree = renderer
      .create(<Counter count={count} updateCount={setCount} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
