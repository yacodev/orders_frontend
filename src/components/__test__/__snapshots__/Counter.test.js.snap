// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
<div
  className="css-okb5hw"
>
  <button
    className="css-8th4lx"
    onClick={[Function]}
  >
     - 
  </button>
  <p
    className="css-1u7p27y"
  >
     
    1
     
  </p>
  <button
    className="css-8th4lx"
    onClick={[Function]}
  >
     + 
  </button>
</div>
`;
